stdlib/asm/mem.masm

```diff
@@ -0,0 +1,41 @@
+# ===== MEMORY FUNCTIONS ==========================================================================
+
+#! memcopy(read_ptr: address, write_ptr: address, words: felt)
+#!
```

Comment:
nit: I'd probably get rid of this line as it raises question about how things are arranged on the stack. In my opinion, stack transition diagram described below is sufficient.

another minor nit: I'd probably use `n` instead of `words` to indicate the number of words to copy.